### ISSUE TITLE
New Request: cloudflare.com/learning/

### ISSUE THREAD

**Issue by Rexadev** [openzim/zim-requests#530]
- Website URL: https://www.cloudflare.com/learning/
- License: 
- Desired ZIM Title: CloudFlare Learning
- Desired ZIM Description: 
- Desired ZIM Icon –png (URL or attach one): 
- Language (ISO 639-3): **eng**
- Is this a MediaWiki?: **no**

**Comment by Rexadev** [comment]
@Popolechien Please make this zim

**Comment by RavanJAltaie** [comment]
Recipe is created, labeled for review, will update here once succeeded or failed
https://farm.openzim.org/recipes/cloudflare_en_all

**Comment by Rexadev** [comment]
@RavanJAltaie updates

**Comment by Rexadev** [comment]
@Rexadev updates ?

**Comment by RavanJAltaie** [comment]
Recipe failed.

**Comment by Rexadev** [comment]
1. https://farm.openzim.org/recipes/cloudflare_en_all says succussed 1 month ago.
2. Are you dropping this site

@RavanJAltaie

**Comment by Rexadev** [comment]
https://farm.openzim.org/recipes/cloudflare_en_all says succussed 1 month ago.
Are you dropping this site
@RavanJAltaie

**Comment by RavanJAltaie** [comment]
The recipe succeeded earlier but the output file is not complete.

**Comment by Rexadev** [comment]
@RavanJAltaie why is this closed

**Comment by Popolechien** [comment]
@Rexadev Because the recipe has been created. The dependency is on zimit now where the bug is. We will re-run it when the tool is updated but there is nothing else we can do for the time being.

**Comment by Rexadev** [comment]
@Popolechien Recipe succeeded once please upload that to library

**Comment by Rexadev** [comment]
@RavanJAltaie Updates

**Comment by benoit74** [comment]
I fixed the recipe (moved URL from `https://www.cloudflare.com/learning` to `https://www.cloudflare.com/learning/` so that prefix scope works correctly and we do not scrape the whole Cloudflare website), migrated it to Zimit 2.0.1 and relaunched the recipe.

Hopefully it will do the job. A custom CSS would help to hide everything which is not going to be offlined.

**Comment by benoit74** [comment]
I also fixed the recipe name and ZIM name (from `cloudflare_en_learning` to `cloudflare.com_en_learning-center`). Recipe is now at https://farm.openzim.org/recipes/cloudflare.com_en_learning-center.

**Comment by benoit74** [comment]
@RavanJAltaie could you please have a look, the ZIM seems pretty ok to me: https://dev.library.kiwix.org/viewer#cloudflare.com_en_learning-center_2024-06. I recommend to hide some stuff with custom, but other than that it seems the ZIM is pretty good.

**Comment by benoit74** [comment]
@RavanJAltaie all these links are not part of the cloudflare learning center, so they are excluded on purpose since this request is about the learning center. Do you still want to ZIM them? I think this header and the footer of the page should be hidden since they contains mostly only broken links not related to the learning center.

**Comment by RavanJAltaie** [comment]
@benoit74 I don't want to ZIM them but let's hide them because anyone who will use the file will click on them and will think they are a part of the file which don't open or work.

**Comment by benoit74** [comment]
Custom CSS has been written and recipe reconfigured and requested.

Developing the custom CSS I realized that learning center content is in fact available in many more languages:
![Screenshot 2024-07-01 at 09 10 30](https://github.com/openzim/zim-requests/assets/7102089/4f8fbbe9-f1e9-401f-afce-8b629b72b2ca)

These other languages are not in current ZIM (which is hence correct) but I think we should duplicate the recipe to have on ZIM per language available to increase availability of this content. @RavanJAltaie can you do that please?

**Comment by benoit74** [comment]
custom CSS was insufficient, not coverage all banners Cloudflare tries to add, so I fixed this and restarted the recipe

**Comment by RavanJAltaie** [comment]
> Custom CSS has been written and recipe reconfigured and requested.
> 
> Developing the custom CSS I realized that learning center content is in fact available in many more languages: ![Screenshot 2024-07-01 at 09 10 30](https://private-user-images.githubusercontent.com/7102089/344578616-4f8fbbe9-f1e9-401f-afce-8b629b72b2ca.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5Mjk2NTcsIm5iZiI6MTcxOTkyOTM1NywicGF0aCI6Ii83MTAyMDg5LzM0NDU3ODYxNi00ZjhmYmJlOS1mMWU5LTQwMWYtYWZjZS04YjYyOWI3MmIyY2EucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MDJUMTQwOTE3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWFjZmJiMjc3YzQxNWIzY2ZjNzhiMzlmMWJlZjUyNjZkYjFlZmU3ZDBiZGMyMzVjMTYwMmVhOWNlMGFmNTE5YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.3KYASMLSdw644vlRa00CNmFuqmUxrpHFRhC--hNrp5s)
> 
> These other languages are not in current ZIM (which is hence correct) but I think we should duplicate the recipe to have on ZIM per language available to increase availability of this content. @RavanJAltaie can you do that please?

I can do that, are we expecting a zim file for each of the available languages? @benoit74 @Popolechien

**Comment by benoit74** [comment]
Agreed regarding the one recipe / one ZIM per available language, there is no reason / interest / constraint in creating a `mul` ZIM here.

Regarding `en`, I let you check https://dev.library.kiwix.org/viewer#cloudflare.com_en_learning-center_2024-07, I think you can now move the recipe to production, I let you decide / do this

**Comment by benoit74** [comment]
PS: I think we need only one ZIM per language, i.e. we do not mind about the English and Espanol locales (UK, Canada, ...), we just ZIM the most "popular" locale.

**Comment by RavanJAltaie** [comment]
@benoit74 Noted.
